src/line2d.jl

```diff
@@ -1,84 +1,73 @@
-abstract Line
 
-immutable LineTwoPoints <: Line
-    y0::Number
-    x0::Number
-    y1::Number
-    x1::Number
-end
-
-LineTwoPoints(p1::CartesianIndex{2}, p2::CartesianIndex{2}) = LineTwoPoints(p1[1],p1[2],p2[1],p2[2])
+#Function to return valid intersections of lines with image boundary
 
-immutable LineNormal <: Line
-    ρ::Number
-    θ::Number
+function get_valid_intersections(intersections::Vector{Tuple{Real, Real}}, indsx::Base.OneTo{Int}, indsy::Base.OneTo{Int})
```

Comment:
signature on `indsx` and `indsy` is too restrictive, just make them `AbstractUnitRange` or something. Note that `first(::OneTo) == 1`, so there was a mismatch of generality here.